src/controller/CHIPCommissionableNodeController.cpp

```diff
@@ -41,5 +46,22 @@ const Mdns::DiscoveredNodeData * CommissionableNodeController::GetDiscoveredComm
     return GetDiscoveredNode(idx);
 }
 
+CHIP_ERROR CommissionableNodeController::AdvertiseCommissionableNode()
+{
+#if CONFIG_DEVICE_LAYER
+    ReturnErrorOnFailure(chip::Mdns::ServiceAdvertiser::Instance().Start(&chip::DeviceLayer::InetLayer, chip::Mdns::kMdnsPort));
+    return app::Mdns::AdvertiseCommissionableNode();
+#else
+    return CHIP_ERROR_NOT_IMPLEMENTED;
+#endif
+}
+
+CHIP_ERROR CommissionableNodeController::SendUserDirectedCommissioningRequest(chip::Inet::IPAddress commissioner, uint16_t port)
```

Comment:
Where would the input to this function come from?

Reply:
It would come from the user in the form of a command line argument to this command as per step # 5 in https://github.com/CHIP-Specifications/connectedhomeip-spec/blob/master/src/rendezvous/InitiatingSetup.adoc#242-high-level-flow (in a follow-up PR).